src/node/ledgersecrets.h

```diff
@@ -153,14 +159,16 @@ namespace ccf
       auto search = secrets_map.find(new_v);
       if (search != secrets_map.end())
       {
-        LOG_FAIL_FMT("promote_secrets(): secrets already exist");
+        LOG_FAIL_FMT(
+          "Cannot promote ledger secret {} as they already exist", new_v);
         return false;
       }
 
       search = secrets_map.find(old_v);
       if (search == secrets_map.end())
       {
-        LOG_FAIL_FMT("promote_secrets(): no secrets to promote");
+        LOG_FAIL_FMT(
+          "Cannot promote ledger secret {} as they do not exist", old_v);
```

Comment:
This message and the one above are unclear.

"Cannot promote ledger secrets at {} - secrets at this version already exist and would be overwritten"
"Cannot promote ledger secrets from {} - secrets do not exist from this version"

Not sure if this should be "secret" or "secrets".